Fix comparison that was not properly checking for option

mikeyarce opened the pull request:
Fixes #5672

### Changes proposed in this Pull Request

Right now the plugins is comparing a **string** of `-1` with an integer of `-1` which means that any time the option is set to anything it will update, causing a bug.

Would it be better to change the check to a string / stirng comparison, or should I cast an integer to make it an int / int comparison?  In this PR I chose the string / string comparison.

### Testing instructions

- Go to a Lesson
- Enable "Random Question Order"
- Click "Update"
- Disable "Random Question Order"
- Click "Update"
- Click "Update" again without changing anything
- Refresh the page
- Check that "Random Quesion Order" is NOT set.

Edit:  I think maybe we should also check the rest of the comparisons in this file and make it more type friendly?

mikeyarce committed:
Fix comparison that was not properly checking for option

dadish commented:
It's weird that we are using `-1` as a placeholder for empty values. This issue relates to all the quiz settings. It comes from here https://github.com/Automattic/sensei/blob/6a24404130b153a6109297262d62e744dbf9f551/includes/class-sensei-lesson.php#L674-L690

I agree, we should update the rest of the comparisons so they use `'-1'` string as a comparison instead of `-1` number.

You can also refactor the above lines to use `null` instead of `'-1'` string. Either ways is fine.

yscik commented:
Casting it to `int` covers a bit more cases. But I agree that those would be better as `null` for empty values. Just have to make there is no logic where an empty/null value would instead trigger some other default? And also we'd still need to support the existing `-1`s already in the DB.

mikeyarce committed:
Change -1 to null

mikeyarce commented:
Thanks for your input @yscik and @dadish! 

I changed the values to `null` and tested the following options:
- pass_required
- quiz_passmark
- enable_quiz_reset
- show_questions
- random_question_order
- quiz_grade_type

They all had no problems and now work as expected.   No need to worry about previous DB values because none of them are `-1`, we're just using that as a check to see if they are set or not.

mikeyarce committed:
Add check for block editor

This currently runs on the Block Editor but it’s legacy code that should only run on the Classic Editor.

mikeyarce commented:
Added a check to make sure this saving doesn't happen in the Block Editor.  Currently, it was running in the Block Editor which was creating two saves happening which could lead to unintended bugs.

mikeyarce committed:
Fix PHP Lint

dadish reviewed (dismissed):
@mikeyarce Looks good.

Could you please fix the "PHP Linting and Tests / PHP Unit Tests..." checks before merging.

mikeyarce committed:
Fix unit test

Manually set the Block Editor check to false so that meta_box_save can run properly.

mikeyarce committed:
Fix return function
